test/local/account_routes.js

```diff
@@ -760,607 +514,164 @@ test(
       t.equal(eventData.data.metricsContext.entrypoint, 'preferences', 'it contained the entrypoint metrics field')
       t.equal(eventData.data.metricsContext.utm_content, 'some-content-string', 'it contained the utm_content metrics field')
-    }).finally(function () {
-      mockLog.close()
-    })
-  }
-)
-
-test(
-  'login with disabled sign-in confirmation, sends new device email',
-  function (t) {
-    var configOptions = {
-      signinConfirmation: {
-        enabled: false,
-        supportedClients: ['fx_desktop_v3'],
-        forceEmailRegex:['mozilla.com$']
-      },
-      newLoginNotificationEnabled: true
-    }
 
-    var uid = uuid.v4('binary')
-    var mockRequest = mocks.mockRequest(TEST_EMAIL, 'true')
-    var mockDB = mocks.mockDB(uid, TEST_EMAIL, true)
-    var mockMailer = mocks.mockMailer()
-
-    var accountRoutes = makeRoutes({
-      config: configOptions,
-      db: mockDB,
-      mailer: mockMailer,
-      checkPassword: function () {
-        return P.resolve(true)
-      }
-    })
+      t.equal(mockLog.metricsContext.validate.callCount, 1, 'metricsContext.validate was called')
+      var call = mockLog.metricsContext.validate.getCall(0)
+      t.equal(call.args.length, 1, 'validate was called with a single argument')
+      t.deepEqual(call.args[0], mockRequest, 'validate was called with the request')
 
-    return new P(function (resolve) {
-      getRoute(accountRoutes, '/account/login')
-        .handler(mockRequest, function (response) {
-          resolve(response)
-        })
+      t.equal(mockMailer.sendNewDeviceLoginNotification.callCount, 1, 'mailer.sendNewDeviceLoginNotification was called')
+      t.equal(mockMailer.sendVerifyLoginEmail.callCount, 0, 'mailer.sendVerifyLoginEmail was not called')
+      t.notOk(response.verificationMethod, 'verificationMethod doesn\'t exist')
+      t.notOk(response.verificationReason, 'verificationReason doesn\'t exist')
+    }).then(function () {
+      mockMailer.sendNewDeviceLoginNotification.reset()
     })
-      .then(function (response) {
-        t.equal(mockMailer.sendNewDeviceLoginNotification.callCount, 1, 'mailer.sendNewDeviceLoginNotification was called')
-        t.equal(mockMailer.sendVerifyLoginEmail.callCount, 0, 'mailer.sendVerifyLoginEmail was not called')
-        t.notOk(response.verificationMethod, 'verificationMethod doesn\'t exist')
-        t.notOk(response.verificationReason, 'verificationReason doesn\'t exist')
-      })
-  }
-)
-
-test(
-  'login with enabled sign-in confirmation',
-  function (t) {
-    var configOptions = {
-      signinConfirmation: {
-        enabled: true,
-        sample_rate: 1.0,
-        supportedClients: ['fx_desktop_v3'],
-        forceEmailRegex:['mozilla.com$']
-      }
+  }, t)
+
+  test('sign-in confirmation enabled', function (t) {
+    config.signinConfirmation = {
+      enabled: true,
+      supportedClients: [ 'fx_desktop_v3' ],
+      forceEmailRegex: [ '.+@mozilla\.com$', 'fennec@fire.fox' ]
```

Comment:
I like this much better than what I had. Do you want to update `'.+@mozilla\.com$'` in the config as well? https://github.com/mozilla/fxa-auth-server/blob/master/config/index.js#L426